src/ip.py

```diff
@@ -125,6 +126,99 @@ def randomMAC():
     return ":".join(["%02x" % x for x in mac])
 
 
+# Decode ifconfig output
+def parse_ifconfig(res, address):
+    links = []
+    ifindex = 1
+
+    for r in res.split("\n"):
+        if re.match(r"^\w+:", r):
+            if ifindex > 1:
+                links.append(link)
+            (ifname, flags, mtu) = re.findall(r"^(\w+): flags=\d+<(.*)> mtu (\d+)", r)[0]
+            flags = flags.split(",")
+            link = {
+                "ifindex": ifindex,
+                "ifname": ifname,
+                "flags": flags,
+                "mtu": mtu
+            }
+            if "LOOPBACK" in flags:
+                link["link_type"] = "loopback"
+                link["address"] = "00:00:00:00:00:00"
+                link["broadcast"] = "00:00:00:00:00:00"
+            elif "POINTOPOINT" in flags:
+                link["link_type"] = "none"
+            else:
+                link["link_type"] = "unknown"
+            ifindex = ifindex + 1
+        else:
+            if re.match(r"^\s+ether ", r):
+                link["link_type"] = "ether"
+                link["address"] = re.findall(r"(\w\w:\w\w:\w\w:\w\w:\w\w:\w\w)", r)[0]
+                link["broadcast"] = "ff:ff:ff:ff:ff:ff"
+            elif address and re.match(r"^\s+inet ", r):
+                (local, netmask) = re.findall(r"inet (\d+\.\d+\.\d+\.\d+) netmask (0x[0-9a-f]+)", r)[0]
+                addr = {
+                    "family": "inet",
+                    "local": local,
+                    "prefixlen": netmask_to_length(netmask),
+                }
+                if re.match(r"^.*broadcast", r):
+                    addr["broadcast"] = re.findall(r"broadcast (\d+\.\d+\.\d+\.\d+)", r)[0]
+                link["addr_info"] = link.get("addr_info", []) + [addr]
+            elif address and re.match(r"^\s+inet6 ", r):
+                (local, prefixlen) = re.findall(r"inet6 ([0-9a-f:]*::[0-9a-f:]+)%*\w* prefixlen (\d+)", r)[0]
+                link["addr_info"] = link.get("addr_info", []) + [{
+                  "family": "inet6",
+                  "local": local,
+                  "prefixlen": int(prefixlen)
+                }]
+
+    links.append(link)
+
+    return links
+
+
+def link_addr_show(argv, af, address):
```

Comment:
Both `ip link show` and `ip addr show` parse the same output from `ifconfig`.
This function collect and parse it.

Reply:
I understand that these are now much more linux-like, but it doesn't print ifindex.

Did you look into what information might be missing? And if we might need to implement some extra flags to show some of the details?

For example, old output had:

```
...
bridge0: flags=8863<UP,BROADCAST,SMART,RUNNING,SIMPLEX,MULTICAST> mtu 1500
	options=63<RXCSUM,TXCSUM,TSO4,TSO6>
	ether fa:ke:ad:dr:es:ss
	Configuration:
		id 0:0:0:0:0:0 priority 0 hellotime 0 fwddelay 0
		maxage 0 holdcnt 0 proto stp maxaddr 100 timeout 1200
		root id 0:0:0:0:0:0 priority 0 ifcost 0 port 0
		ipfilter disabled flags 0x0
	member: en1 flags=3<LEARNING,DISCOVER>
	        ifmaxaddr 0 port 9 priority 0 path cost 0
	member: en2 flags=3<LEARNING,DISCOVER>
	        ifmaxaddr 0 port 10 priority 0 path cost 0
	nd6 options=201<PERFORMNUD,DAD>
	media: <unknown type>
	status: inactive
...
```


New output has only:
```
...
bridge0: <UP,BROADCAST,SMART,RUNNING,SIMPLEX,MULTICAST> mtu 1500
    link/ether fa:ke:ad:dr:es:ss brd ff:ff:ff:ff:ff:ff
...
```

I would still rather prefer compatibility with linux output, but I wonder if we should consider and or document that there will be no longer details in the output with the new version.

FYI `ip -details -pretty -json link show type bridge` show quite some details for bridge interfaces.

Reply:
Thanks to `ifconfig -v` output now ifindex is printed consistently.

A `-details` flag is absolutely desirable. It implies a collection of more informations and filter them in a second stage. I'll try to work on it in a few days.

Reply:
Moreover `ip` command from `iproute2` does not show bridges information like slaves (as macOS `ifconfig` do) even with `-details` option. Instead, it relies on `bridge` command, supplied within `iproute2` package to work on bridges.

Should be a `bridge.py` command be bundled within `iproute2mac` to enhance compatibility with Linux commands and cover lacks of `ip` output?

Reply:
I ended up to implement a read only version of `bridge` command to show bridged interfaces:

```
# ip addr show dev bridge0
14: bridge0: <UP,BROADCAST,SMART,RUNNING,SIMPLEX,MULTICAST> mtu 1500 status DOWN
    link/ether 36:41:94:18:66:40 brd ff:ff:ff:ff:ff:ff
```

```
# bridge link show
9: en2: <UP,BROADCAST,SMART,RUNNING,PROMISC,SIMPLEX,MULTICAST> mtu 1500 master bridge0 state forwarding priority 0 cost 0
10: en3: <UP,BROADCAST,SMART,RUNNING,PROMISC,SIMPLEX,MULTICAST> mtu 1500 master bridge0 state forwarding priority 0 cost 0
```